<a href="https://colab.research.google.com/github/elisaramoss/analise_BanVic/blob/main/Desafio_de_Engenharia_de_Analytics_An%C3%A1lise_de_Dados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Desafio de Engenharia de Analytics/Análise de Dados 2024-9


Neste desafio você vai gerar um relatório com indicadores de performance para responder às demandas do BanVic. Para isso você deve criar indicadores de negócio e sugerir como a empresa pode agir em cima deles para melhorar seus resultados. Você pode usar as ferramentas que tiver maior facilidade ou achar mais interessante tais como Excel, Google Sheets, Python, R, dbt (data build tool) etc.  Lembre que este relatório será apresentado para a CEO, então você deve formatá-lo de forma profissional e objetiva.


##Agências

In [2]:
#Importando os pacotes necessários
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [25]:
#Carregar os dados da Agência BanVic

df_agencia = pd.read_csv("https://raw.githubusercontent.com/elisaramoss/projetos-bootcamp/main/banvic_data/agencias.csv")
df_agencia

cod_agencia                    nome  \
0            7         Agência Digital   
1            1          Agência Matriz   
2            2         Agência Tatuapé   
3            3        Agência Campinas   
4            4          Agência Osasco   
5            5    Agência Porto Alegre   
6            6  Agência Rio de Janeiro   
7            8         Agência Jardins   
8            9   Agência Florianópolis   
9           10          Agência Recife   

                                            endereco          cidade  uf  \
0  Av. Paulista, 1436 - Cerqueira César, São Paul...       São Paulo  SP   
1  Av. Paulista, 1436 - Cerqueira César, São Paul...       São Paulo  SP   
2  Praça Sílvio Romero, 158 - Tatuapé, São Paulo ...       São Paulo  SP   
3  Av. Francisco Glicério, 895 - Vila Lidia, Camp...        Campinas  SP   
4  Av. Antônio Carlos Costa, 1000 - Bela Vista, O...          Osasco  SP   
5  Av. Bento Gonçalves, 1924 - Partenon, Porto Al...    Porto Alegre  RS   
6  R. Sen. Dantas, 15 - Centro, Rio de Janeiro - ...  Rio de Janeiro  RJ   
7  Av. Brg. Faria Lima, 2491 - Jardim Paulistano,...       São Paulo  SP   
8  Av. Jorn. Rubéns de Arruda Ramos, 1280 - Centr...   Florianópolis  SC   
9  Av. Conselheiro Aguiar, 4432 - Boa Viagem, Rec...          Recife  PE   

  data_abertura tipo_agencia  
0    2015-08-01      Digital  
1    2010-01-01       Física  
2    2010-06-14       Física  
3    2012-03-04       Física  
4    2013-11-06       Física  
5    2013-12-01       Física  
6    2015-04-01       Física  
7    2018-01-09       Física  
8    2019-10-09       Física  
9    2021-10-09       Física

In [4]:
# Analisei as distribuições de região
agencia_fisica = df_agencia[df_agencia['tipo_agencia'] == 'Física']
agencia_por_regiao =agencia_fisica['uf'].value_counts(normalize=True)

print(f"Quantidade de Agências: {len(df_agencia)}")

#Porcentagem das agencias físicas po uf
print(f"Porcentagem de agências físicas por {agencia_por_regiao}")

Quantidade de Agências: 10
Porcentagem de agências físicas por uf
SP    0.555556
RS    0.111111
RJ    0.111111
SC    0.111111
PE    0.111111
Name: proportion, dtype: float64


A partir da análise das distribuições das agências do Banco pelo Brasil, percebe-se que mais de 50% das agências físicas se encontram no Estado de São Paulo.

###Tratamento do dados Agências


In [26]:
# Separando o endereço em logradouro, número de logradouro e bairro.
# Cidade e UF já possuem colunas próprias
df_agencia['logradouro'] = df_agencia.endereco.apply(lambda x: x.split(',')[0])
df_agencia['no_logradouro'] = df_agencia.endereco.apply(lambda x: x.split(',')[1].split('-')[0])
df_agencia['bairro'] = df_agencia.endereco.apply(lambda x: x.split(',')[1].split('-')[1])
df_agencia = df_agencia.drop('endereco', axis=1)
df_agencia['nome_completo'] = df_agencia.nome + ' - ' + df_agencia.cidade + '/' + df_agencia.uf
df_agencia



cod_agencia                    nome          cidade  uf data_abertura  \
0            7         Agência Digital       São Paulo  SP    2015-08-01   
1            1          Agência Matriz       São Paulo  SP    2010-01-01   
2            2         Agência Tatuapé       São Paulo  SP    2010-06-14   
3            3        Agência Campinas        Campinas  SP    2012-03-04   
4            4          Agência Osasco          Osasco  SP    2013-11-06   
5            5    Agência Porto Alegre    Porto Alegre  RS    2013-12-01   
6            6  Agência Rio de Janeiro  Rio de Janeiro  RJ    2015-04-01   
7            8         Agência Jardins       São Paulo  SP    2018-01-09   
8            9   Agência Florianópolis   Florianópolis  SC    2019-10-09   
9           10          Agência Recife          Recife  PE    2021-10-09   

  tipo_agencia                        logradouro no_logradouro  \
0      Digital                      Av. Paulista         1436    
1       Física                      Av. Paulista         1436    
2       Física               Praça Sílvio Romero          158    
3       Física            Av. Francisco Glicério          895    
4       Física          Av. Antônio Carlos Costa         1000    
5       Física               Av. Bento Gonçalves         1924    
6       Física                    R. Sen. Dantas           15    
7       Física               Av. Brg. Faria Lima         2491    
8       Física  Av. Jorn. Rubéns de Arruda Ramos         1280    
9       Física            Av. Conselheiro Aguiar         4432    

               bairro                               nome_completo  
0     Cerqueira César              Agência Digital - São Paulo/SP  
1     Cerqueira César               Agência Matriz - São Paulo/SP  
2             Tatuapé              Agência Tatuapé - São Paulo/SP  
3          Vila Lidia              Agência Campinas - Campinas/SP  
4          Bela Vista                  Agência Osasco - Osasco/SP  
5            Partenon      Agência Porto Alegre - Porto Alegre/RS  
6              Centro  Agência Rio de Janeiro - Rio de Janeiro/RJ  
7   Jardim Paulistano              Agência Jardins - São Paulo/SP  
8              Centro    Agência Florianópolis - Florianópolis/SC  
9          Boa Viagem                  Agência Recife - Recife/PE

##Clientes

In [ ]:
#Carregando os dados dos clientes do banco

df_clientes = pd.read_csv("https://raw.githubusercontent.com/elisaramoss/projetos-bootcamp/main/banvic_data/clientes.csv")

print(f"Quantidade de clientes: {df_clientes.shape[0]}")

df_clientes.head(10)

Quantidade de clientes: 998


cod_cliente  primeiro_nome ultimo_nome                          email  \
0           28        Sabrina        Dias   moreiraemanuelly@example.org   
1          674    Luiz Felipe        Dias      pedroferreira@example.org   
2          693          Renan        Dias             ogomes@example.net   
3          743          Clara        Dias      rafaelcorreia@example.org   
4          769          André        Dias           danilo33@example.org   
5          824        Heloísa        Dias          alanadias@example.org   
6          884    Maria Julia        Dias       alvesmariana@example.net   
7          899         Arthur        Dias  castropedro-lucas@example.net   
8          908  Vitor Gabriel        Dias           acardoso@example.com   
9          945           Caio        Dias       moraesmarina@example.org   

  tipo_cliente            data_inclusao         cpfcnpj data_nascimento  \
0           PF  2017-04-03 16:11:00 UTC  357.081.496-39      2006-08-11   
1           PF  2021-02-10 13:27:00 UTC  085.196.374-93      1995-10-11   
2           PF  2020-01-21 13:12:00 UTC  783.416.059-10      1948-11-19   
3           PF  2019-05-06 11:39:00 UTC  589.237.610-95      1978-01-27   
4           PF  2017-01-07 14:53:00 UTC  459.608.721-02      1990-08-25   
5           PF  2016-01-07 12:26:00 UTC  536.981.240-33      1967-05-12   
6           PF  2017-08-06 11:01:00 UTC  150.438.926-33      1950-12-10   
7           PF  2021-10-17 13:00:00 UTC  509.481.762-02      1953-02-26   
8           PF  2021-04-02 10:22:00 UTC  158.406.379-39      1948-03-25   
9           PF  2016-05-10 13:56:00 UTC  975.860.431-75      1955-06-11   

                                            endereco        cep  
0  Praia de Duarte Vila Piratininga 81327-166 Fer...  95140-704  
1  Avenida da Rosa, 654 João Paulo Ii 20295449 Nu...  76516-765  
2  Jardim de Rodrigues Ipiranga 14161-477 Duarte ...   51779625  
3  Colônia Thomas Silva, 9 Tupi B 15771-946 Ferna...   19615792  
4     Rua Correia, 889 Diamante 59123250 Aragão / RS   01672838  
5  Favela de da Mata, 3 Conjunto Novo Dom Bosco 7...  07814-131  
6  Esplanada Maria Luiza Ribeiro, 4 Conjunto Cali...  90221-380  
7  Ladeira Elisa da Rocha, 69 João Pinheiro 13964...  44875-244  
8  Passarela Lucca Castro, 73 Vila Paris 99268696...  42603-833  
9  Fazenda Emanuel Gomes, 6 Cruzeiro 29428-910 No...   23176352

#Colaboradores

In [30]:
#carregando os dados dos colaboradores

df_colaboradores = pd.read_csv("https://raw.githubusercontent.com/elisaramoss/projetos-bootcamp/main/banvic_data/colaboradores.csv")

print(f"Quantidade de colaboradores: {df_colaboradores.shape[0]}")
df_colaboradores.head(10)

Quantidade de colaboradores: 100


cod_colaborador  primeiro_nome ultimo_nome                           email  \
0               14          Paulo        Dias        melissalopes@example.net   
1               19  Luiz Fernando        Dias              pcunha@example.net   
2               32     Vitor Hugo        Dias       aragaonicolas@example.net   
3               88   João Gabriel        Lima  luiz-otaviorezende@example.org   
4               39  Maria Eduarda        Melo        joao-pedro72@example.com   
5               41         Olivia       Costa           sabrina59@example.com   
6               45          Laura       Cunha              lviana@example.org   
7              100        Lavínia       Cunha          noahmoraes@example.com   
8               16       Benjamin       Gomes              vjesus@example.org   
9               46          Alana       Gomes            samuel76@example.com   

              cpf data_nascimento  \
0  847.210.695-02      1974-04-24   
1  820.415.963-33      1994-07-10   
2  936.172.548-37      1986-12-18   
3  967.082.451-67      1994-10-19   
4  857.496.210-49      1996-03-01   
5  190.768.435-20      1966-01-30   
6  217.583.096-95      1977-02-07   
7  062.937.158-02      1975-03-24   
8  718.950.436-20      1963-12-21   
9  768.134.950-20      1977-07-16   

                                            endereco        cep  
0  Lagoa de Rodrigues, 62 Zilah Sposito 37328-273...   27275674  
1  Ladeira de Moreira, 82 Jardim Atlântico 27617-...   50013962  
2  Área Sophie Moreira Maria Virgínia 71941-111 S...  08955-215  
3  Trevo Murilo Ramos, 92 São Benedito 73615-597 ...  24224-305  
4  Parque de Moura, 65 Granja Werneck 67769059 da...   96896390  
5   Favela Rocha, 94 Laranjeiras 92500120 Viana / GO   90019993  
6  Lagoa Benício Moura, 5 Vila Paraíso 50235854 d...  19415-341  
7   Colônia Lima, 83 Flamengo 36172-665 da Mota / SE   74890752  
8  Conjunto Gomes, 888 Céu Azul 76308-427 Lopes d...   29604844  
9  Fazenda Emilly Costa Campo Alegre 55916-800 da...  88161-168

#Colaborador por Agência

In [ ]:
#Carregando os dados dos colaboradores por agência

df_colaborador_agencia = pd.read_csv("https://raw.githubusercontent.com/elisaramoss/projetos-bootcamp/main/banvic_data/colaborador_agencia.csv")

print(f"Quantidade de colaboradores por agência: {df_colaborador_agencia.shape[0]}")

# Contando o número de colaboradores por agência
colaboradores_por_agencia = df_colaborador_agencia.groupby('cod_agencia').size()

print("\n")
# Exibindo a contagem de colaboradores por agência
print(f"Quantidade de colaboradores por agência:")
print(colaboradores_por_agencia)
df_colaborador_agencia.head(10)


Quantidade de colaboradores por agência: 100


Quantidade de colaboradores por agência:
cod_agencia
1     16
2     13
3     11
4      4
5     13
6     11
7      7
8      7
9      8
10    10
dtype: int64


cod_colaborador  cod_agencia
0                7            1
1               13            1
2               14            1
3               23            1
4               24            1
5               30            1
6               31            1
7               34            1
8               42            1
9               58            1

Interprentando a descrição acima, infere-se que a Agência Matriz em SP possui o maior número de colaboradores,em seguida as agências de Tatuapé em SP e Porto Alegre	RS.

#Contas

In [32]:
#Carregando os dados das contas dos clientes

df_contas = pd.read_csv("https://raw.githubusercontent.com/elisaramoss/projetos-bootcamp/main/banvic_data/contas.csv")

print(f"Quantidade de contas: {df_contas.shape[0]}")

df_contas.head(10)

Quantidade de contas: 999


num_conta  cod_cliente  cod_agencia  cod_colaborador tipo_conta  \
0         53           53            1                7         PF   
1        190          190            1                7         PF   
2        191          191            1                7         PF   
3        203          203            1                7         PF   
4        254          254            1                7         PF   
5        306          306            1                7         PF   
6        472          472            1                7         PF   
7        520          520            1                7         PF   
8        739          739            1                7         PF   
9        509          509            1               13         PF   

             data_abertura  saldo_total  saldo_disponivel  \
0  2011-05-24 15:30:00 UTC    2984.7615           2814.67   
1  2014-10-17 14:48:00 UTC    4262.4322           4191.86   
2  2013-03-23 14:14:00 UTC    3266.5680           3080.71   
3  2016-08-03 16:13:00 UTC   21241.5424          19461.56   
4  2012-10-24 13:48:00 UTC   13335.6966          13197.54   
5  2012-05-23 12:11:00 UTC   22997.4738          21677.88   
6  2022-08-07 12:08:00 UTC    4178.4237           4183.68   
7  2019-11-03 16:26:00 UTC    1809.8087           1288.51   
8  2014-02-26 12:20:00 UTC      45.7691            616.79   
9  2012-01-20 11:56:00 UTC   19159.7952          19863.76   

           data_ultimo_lancamento  
0         2019-07-02 11:03:46 UTC  
1         2021-03-25 12:00:15 UTC  
2         2016-09-09 23:34:15 UTC  
3         2022-12-28 07:58:24 UTC  
4         2013-12-20 22:20:00 UTC  
5         2016-10-05 03:14:36 UTC  
6   2022-12-30 00:00:00.67516 UTC  
7  2022-12-29 23:59:59.753932 UTC  
8  2022-12-29 23:59:59.789361 UTC  
9         2016-06-09 22:25:04 UTC

Como foi observado anteriormente existem apenas 988 clientes, mas o banco possui 999 contas, dessa forma, podemos inferir que pelo menos um cliente há mais de uma conta cadastrada.

#Propostas de Crédito

In [33]:
#carregando os dados de propostas de crédito

df_credito = pd.read_csv("https://raw.githubusercontent.com/elisaramoss/projetos-bootcamp/main/banvic_data/propostas_credito.csv")

print(f"Quantidade de propostas de crédito: {df_credito.shape[0]}")

df_credito.head(10)

Quantidade de propostas de crédito: 2000


cod_proposta  cod_cliente  cod_colaborador    data_entrada_proposta  \
0           116          338                1  2014-05-30 18:43:12 UTC   
1           715           45                1  2021-06-17 13:09:32 UTC   
2           755          494                1  2021-08-10 04:24:02 UTC   
3           953          381                1  2011-02-23 07:57:58 UTC   
4          1046          176                1  2010-11-15 16:03:28 UTC   
5          1171          732                1  2020-05-04 21:51:58 UTC   
6          1434          560                1  2020-01-31 00:02:49 UTC   
7          1744          159                1  2016-03-04 20:31:58 UTC   
8          1864          776                1  2019-09-13 10:21:31 UTC   
9          1928          421                1  2014-05-08 15:43:27 UTC   

   taxa_juros_mensal  valor_proposta  valor_financiamento  valor_entrada  \
0             0.0194    36199.950355             50032.03   13832.079645   
1             0.0131    12897.538285             19848.55    6951.011715   
2             0.0203   122236.312551            187154.36   64918.047449   
3             0.0167    47784.649757             59530.61   11745.960243   
4             0.0211   178393.038243            214294.19   35901.151757   
5             0.0242    40476.959008             56464.88   15987.920992   
6             0.0169   137637.615490            192947.94   55310.324510   
7             0.0180     1317.125762              1605.63     288.504238   
8             0.0110   106127.470610            128472.96   22345.489390   
9             0.0200     3455.606489              6668.98    3213.373511   

   valor_prestacao  quantidade_parcelas  carencia status_proposta  
0      1045.219825                   58         0         Enviada  
1       232.124685                  100         1         Enviada  
2      3673.518951                   56         0         Enviada  
3       955.041560                  109         3         Enviada  
4     10219.659415                   22         4         Enviada  
5     20976.064001                    2         0         Enviada  
6      2861.607605                  100         0         Enviada  
7        36.078665                   60         2         Enviada  
8      1774.933849                   98         4         Enviada  
9       211.333552                   20         2         Enviada

In [34]:
df_credito.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 12 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   cod_proposta           2000 non-null   int64  
 1   cod_cliente            2000 non-null   int64  
 2   cod_colaborador        2000 non-null   int64  
 3   data_entrada_proposta  2000 non-null   object 
 4   taxa_juros_mensal      2000 non-null   float64
 5   valor_proposta         2000 non-null   float64
 6   valor_financiamento    2000 non-null   float64
 7   valor_entrada          2000 non-null   float64
 8   valor_prestacao        2000 non-null   float64
 9   quantidade_parcelas    2000 non-null   int64  
 10  carencia               2000 non-null   int64  
 11  status_proposta        2000 non-null   object 
dtypes: float64(5), int64(5), object(2)
memory usage: 187.6+ KB


In [35]:
#Verificando o status das propostas de crédito
df_credito.status_proposta.value_counts()

status_proposta
Enviada                 527
Aprovada                514
Validação documentos    491
Em análise              468
Name: count, dtype: int64

Tudo BOm?